src/Cake.Core/CakeEnvironment.cs

```diff
@@ -48,6 +51,7 @@ public CakeEnvironment(ICakePlatform platform, ICakeRuntime runtime)
 
             // Get the working directory.
             WorkingDirectory = new DirectoryPath(System.IO.Directory.GetCurrentDirectory());
+            LaunchDirectory = Environment.CurrentDirectory;
```

Comment:
I haven't tested this yet, but I suspect that getting the `Environment.CurrentDirectory` here is too late (as it would reflect the directory where the Cake script is running). If this works, that would mean `WorkingDirectory` would also work for the same purpose, isn't it?

Reply:
Sorry, it's not ready for review because I'm trying to find a way how to test it

Reply:
Maybe an `ILaunchInfo` (with `LaunchDirectory` proprty)  non lazy instance that's registered and injected via DI, that would be as early as possible, singleton(so it wouldn't change) and easy to mock in tests.

Reply:
Little bit confused about how to add ILaunchInfo to FakeEnvironment and test it. Could you please suggest because FakeEnvironment doesn't use DI.

Reply:
@Marusyk You can create a `FakeLaunchInfo` class and add a property to `FakeEnvironment` similar to how it's done for [`Platform`](https://github.com/cake-build/cake/blob/develop/src/Cake.Testing/FakeEnvironment.cs#L35) and [`Runtime`](https://github.com/cake-build/cake/blob/develop/src/Cake.Testing/FakeEnvironment.cs#L44), for example.

Reply:
Thanks a lot. Could you please point me where is the proper place to test it?

Reply:
Given that you're introducing a new property in `src/Cake.Core/CakeEnvironment.cs` the best place to add unit tests would be `src/Cake.Core.Tests/Unit/CakeEnvironmentTests.cs` (which doesn't exist yet, so you'll need to create it)